tests/integration/test_s3_cache_locality/test.py

```diff
@@ -70,26 +71,31 @@ def started_cluster():
 
         yield cluster
     finally:
-        shutil.rmtree(os.path.join(SCRIPT_DIR, "data/generated/"))
+        shutil.rmtree(os.path.join(SCRIPT_DIR, "data/generated/"), ignore_errors=True)
         cluster.shutdown()
 
 
-def check_s3_gets(cluster, node, expected_result, cluster_first, cluster_second, enable_filesystem_cache):
+def check_s3_gets(cluster, node, expected_result, cluster_first, cluster_second, enable_filesystem_cache, lock=False):
     for host in list(cluster.instances.values()):
-        host.query("SYSTEM DROP FILESYSTEM CACHE 'raw_s3_cache'", timeout=30)
+        host.query("SYSTEM DROP FILESYSTEM CACHE 'raw_s3_cache'", ignore_error=True)
+
+    settings = {
+        "enable_filesystem_cache": enable_filesystem_cache,
+        "filesystem_cache_name": "'raw_s3_cache'",
+        }
+
+    if lock:
+        settings["lock_object_storage_task_distribution_ms"] = 30000
 
     query_id_first = str(uuid.uuid4())
     result_first = node.query(
         f"""
         SELECT count(*)
           FROM s3Cluster('{cluster_first}', 'http://minio1:9001/root/data/generated/*', 'minio', '{minio_secret_key}', 'CSV', 'a String, b UInt64')
           WHERE b=42
-        SETTINGS
-          enable_filesystem_cache={enable_filesystem_cache},
-          filesystem_cache_name='raw_s3_cache'
+        SETTINGS {",".join(f"{k}={v}" for k, v in settings.items())}
         """,
         query_id=query_id_first,
-        timeout=30,
```

Comment:
Timeouts removed, because it creates to many descriptors.
(https://github.com/Altinity/ClickHouse/blob/antalya/tests/integration/helpers/client.py#L209)
Locally tests failed with "too many open files".